automates/program_analysis/CAST2GrFN/visitors/cast_to_agraph_visitor.py

```diff
@@ -0,0 +1,449 @@
+        accordingly. The node's UID is returned."""
+        expr = self.visit(node.expr)
+        body = []
+        orelse = []
+        if(node.body != []):
+            body = self.visit_list(node.body) 
+        if(node.orelse != None):
+            orelse = self.visit_list(node.orelse)
+
+        node_uid = uuid.uuid4()
+        test_uid = uuid.uuid4()
+        self.G.add_node(node_uid,label="If")
+        self.G.add_node(test_uid,label="Test")
+        self.G.add_edge(node_uid,test_uid)
+        self.G.add_edge(test_uid,expr)
+        
+        body_uid = uuid.uuid4()
+        orelse_uid = uuid.uuid4()
+
+        self.G.add_node(body_uid,label="If Body")
+        self.G.add_node(orelse_uid,label="Else Body")
+
+        self.G.add_edge(node_uid,body_uid)
+        self.G.add_edge(node_uid,orelse_uid)
+
+        for n in body:
+            self.G.add_edge(body_uid,n)
+
+        for n in orelse:
+            self.G.add_edge(orelse_uid,n)
+
+        return node_uid
+
+    @visit.register
+    def _(self, node: ModelReturn):
+        """Visits a ModelReturn (return statment) node. 
+        We add the return value to this node with an edge.
+        The node's UID is returned"""
+        value = self.visit(node.value)
+        node_uid = uuid.uuid4()
+        self.G.add_node(node_uid, label="Return")
+        self.G.add_edge(node_uid, value)
+
+        return node_uid
+
+    @visit.register
+    def _(self, node: Module):
+        """Visits a Module node. This is the starting point for visiting
+        a CAST object. The return value isn't relevant here (I think)"""
+        return self.visit_list(node.body) 
+
+    @visit.register
+    def _(self, node: Name):
+        """Visits a Name node. We add this node's name to the graph, 
+        and return the UID of this node."""
+        node_uid = uuid.uuid4()
+        self.G.add_node(node_uid,label=node.name)
+        return node_uid 
+
+    @visit.register
+    def _(self, node: Number):
+        """Visits a Number node. We add this node's numeric value to the
+        graph and return the UID of this node."""
+        node_uid = uuid.uuid4()
+        self.G.add_node(node_uid,label=node.number)
+        return node_uid 
+
+    @visit.register
+    def _(self, node: Set):
+        """Visits a Set node. We add all the elements of this set (if any)
+        to the graph and return the UID of this node."""
+        values = []
+        if(node.values != []):
+            values = self.visit_list(node.values)
+        node_uid = uuid.uuid4()
+        self.G.add_node(node_uid,label="Set")
+        for n in values:
+            self.G.add_edge(node_uid,n)
+
+        return node_uid
+
+    @visit.register
+    def _(self, node: String):
+        """Visits a String node. We add this node's string to the
+        graph and return the UID of this node."""
+        node_uid = uuid.uuid4()
+        self.G.add_node(node_uid,label="\""+node.string+"\"")
```

Comment:
small suggestion, you could use pythons built in string formatting in which case you would have 
```
label=f"\"{node.string}\""
```
to achieve the same thing.

Reply:
Feel free to do this as well (`black` will reformat the code like this anyways):
```python
label=f'"{node.string}"'
```

Reply:
Will do! I'm still used to the more complicated (and uglier) method of manually concatenating strings. Thanks!